Assets/Scripts/Map/MapPatrollingGen/WatchmanRouteSolver.cs

```diff
@@ -105,19 +105,19 @@ private static List<Vector2Int> ComputeVertexCoordinates(Dictionary<Vector2Int,
             return guardPositions;
         }
 
-        private static Dictionary<Vector2Int, HashSet<Vector2Int>> ComputeVisibility(bool[,] map, VisibilityMethod visibilityAlgorithm)
+        private static Dictionary<Vector2Int, HashSet<Vector2Int>> ComputeVisibility(BitMap2D map, VisibilityMethod visibilityAlgorithm)
         {
             var precomputedVisibility = new ConcurrentDictionary<Vector2Int, HashSet<Vector2Int>>();
-            var width = map.GetLength(0);
-            var height = map.GetLength(1);
+            var width = map.Width;
+            var height = map.Height;
 
             // Outermost loop parallelized to improve performance
-            Parallel.For(0, width, x =>
+            Parallel.For(0, height, y =>
             {
-                for (var y = 0; y < height; y++)
+                for (var x = 0; x < height; x++)
```

Comment:
```suggestion
                for (var x = 0; x < width; x++)
```
Should it be width instead of height?